### ISSUE TITLE
Two splits may occur when entering MB room

### ISSUE THREAD

**Issue by nortti** [UNHchabo/AutoSplitters#1]
Hi,

Sometimes two splits will occur when entering MB room (NTSC). The first one (the intended one, I believe) occurs at the beginning of the door transition, the second one around halfway through it.

I have most splits enabled, so it's easier to list the ones that are disabled:

- Ammo Pickups
  - All missiles
- Energy Upgrades
  - All reserve tanks
- Misc Upgrades
  - Spring ball
  - Screw Attack
  - Grapple Beam
  - X-Ray Scope
- Minibosses
  - Ceres Ridley
  - Spore Spawn
  - Crocomire
  - Golden Torizo
- Bosses
  - Kraid
  - Draygon
  - Ridley
- IGT Finish
- Spore Spawn RTA Finish
- 100 Missile RTA Finish

I'm using higan 1.06, and I'm not suing any emulator functions. I can help debug this if you cannot reproduce (how would one go about that?).

Thanks for this script btw It's working great otherwise.

**Comment by UNHchabo** [comment]
Sorry for the delay, you filed the issue right as I started a vacation overseas. :)

I've encountered this in the past too, so I should be able to reproduce and debug this, I just haven't had a chance to do it before now.

**Comment by UNHchabo** [comment]
I think I've fixed this.

I was querying MB's health value to split for her different phases -- when it changes from 0 to 18000 that's the start of the second phase (when her head hits the ground), and then when it changes from 0 to 36000 that's the start of the third phase (during the Baby cutscene). During the door transition her HP value usually goes from 0 to 3000, but for some reason sometimes it will go from 0 to 18000 to 3000, which caused the split.

I've changed the Mother Brain splits to query whether we're in "normal gameplay", which means it won't trigger during door transitions. Hopefully this fixes it.
